spec/logger_spec.rb

```diff
@@ -3,29 +3,55 @@
 java_import 'java.util.Properties'
 
 describe Mizuno::Logger do
-  it 'writes logs to a file' do
-    logfile = File.join(File.dirname(__FILE__), '../tmp/logger.log')
-    FileUtils.rm(logfile) if File.exist?(logfile)
-    Mizuno::Logger.configure(log: logfile, debug: true)
-
-    logger = Mizuno::Logger.logger
-    logger.debug('uuwaf')
-    logger.info('shaeg')
-    logger.warn('zohch')
-    logger.error('dooca')
-    logger.fatal('einai')
-
-    content = File.read(logfile).lines.to_a
-    content.grep(/DEBUG uuwaf/).count.should eq 1
-    content.grep(/INFO shaeg/).count.should eq 1
-    content.grep(/WARN zohch/).count.should eq 1
-    content.grep(/ERROR dooca/).count.should eq 1
-    content.grep(/FATAL einai/).count.should eq 1
+  context 'when writing logs to a file' do
+    LOGFILE = File.join(File.dirname(__FILE__), '../tmp/logger.log')
+    let(:logger) { Mizuno::Logger.logger }
+    let(:content) { File.read(LOGFILE).lines.to_a }
+
+    before(:all) {
+      FileUtils.rm(LOGFILE) if File.exist?(LOGFILE)
+      Mizuno::Logger.configure(log: LOGFILE, debug: true)
+    }
+
+    describe '#debug' do
+      it 'prefixes the log message with a DEBUG prefix' do
+        logger.debug('uuwaf')
+        expect(content.grep(/DEBUG uuwaf/).count).to eq 1
+      end
+    end
+
+    describe '#error' do
+      it 'prefixes the log message with a ERROR prefix' do
+        logger.error('dooca')
+        expect(content.grep(/ERROR dooca/).count).to eq 1
+      end
+    end
+
+    describe '#fatal' do
+      it 'prefixes the log message with a FATAL prefix' do
+        logger.fatal('einai')
+        expect(content.grep(/FATAL einai/).count).to eq 1
+      end
+    end
+
+    describe '#info' do
+      it 'prefixes the log message with a INFO prefix' do
+        logger.info('shaeg')
+        expect(content.grep(/INFO shaeg/).count).to eq 1
+      end
+    end
+
+    describe '#warn' do
+      it 'prefixes the log message with a WARN prefix' do
+        logger.warn('zohch')
+        expect(content.grep(/WARN zohch/).count).to eq 1
+      end
+    end
```

Comment:
This is the kind of changes I'm referring to: we should make more of these, so that the specs can be very concise and fine-grained.